lib/rucio/core/config.py

```diff
@@ -423,7 +429,32 @@ def remove_option(section: str, option: str, *, session: "Session") -> bool:
         return True
 
 
-def read_from_cache(key: str, expiration_time: int = 900) -> Any:
+@overload
+def read_from_cache(key: Literal['sections'], expiration_time: int = 900) -> Union[list[str], NoValue]:
+    ...
+
+
+@overload
+def read_from_cache(key: OptionsKey, expiration_time: int = 900) -> Union[list[str], NoValue]:
+    ...
+
+
+@overload
+def read_from_cache(key: HasKey, expiration_time: int = 900) -> Union[bool, NoValue]:
+    ...
+
+
+@overload
+def read_from_cache(key: ItemsKey, expiration_time: int = 900) -> Union[Sequence[Row], NoValue]:
+    ...
+
+
+@overload
+def read_from_cache(key: ValueKey, expiration_time: int = 900) -> Union[Any, NoValue]:
```

Comment:
```
Union[Any, NoValue]
```

Placing `Any` inside an `Union` is redundant: `Any` is compatible with all types https://docs.python.org/3/library/typing.html#typing.Any